### ISSUE TITLE
¿MacOS compilation?

### ISSUE THREAD

**Issue by sergiocasero** [IsmaelMartinez/teams-for-linux#431]
**Is your feature request related to a problem? Please describe.**
Nope

**Describe the solution you'd like**
I like to have a macOS binary

**Describe alternatives you've considered**
Teams web and/or teams native app

**Additional context**
The problem here is that on my company, only windows PCs are allowed, so we can't use MS teams on our macs nor linux. We've tested this repository binaries inside our linux computers and we're able to connect to Teams!! So I suppose that if we could use this electron app also on macos... maybe we can access to. What do you think?

**Comment by IsmaelMartinez** [comment]
Hi, you can get a Mac build no bother. There might be some oddness,
specially on the icon, but in general, it should work.

Ping a PR and/or ask any questions. Happy to help.

**Comment by sergiocasero** [comment]
Hello @IsmaelMartinez ! Sorry for the delay I was on holidays. I'm getting some troubles when trying to build the app, maybe is it related to my node version?? This is the error:

```
error /Users/sergio/projects/teams-for-linux/node_modules/spellchecker: Command failed.
Exit code: 1
Command: node-gyp rebuild
Arguments: 
Directory: /Users/sergio/projects/teams-for-linux/node_modules/spellchecker
Output:
gyp info it worked if it ends with ok
gyp info using node-gyp@7.1.2
gyp info using node@16.0.0 | darwin | x64
gyp info find Python using Python version 3.9.4 found at "/usr/local/opt/python@3.9/bin/python3.9"
(node:7990) [DEP0150] DeprecationWarning: Setting process.config is deprecated. In the future the property will be read-only.
(Use `node --trace-deprecation ...` to show where the warning was created)
gyp info spawn /usr/local/opt/python@3.9/bin/python3.9
gyp info spawn args [
gyp info spawn args   '/usr/local/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
gyp info spawn args   'binding.gyp',
gyp info spawn args   '-f',
gyp info spawn args   'make',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/sergio/projects/teams-for-linux/node_modules/spellchecker/build/config.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/usr/local/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
gyp info spawn args   '-I',
gyp info spawn args   '/Users/sergio/Library/Caches/node-gyp/16.0.0/include/node/common.gypi',
gyp info spawn args   '-Dlibrary=shared_library',
gyp info spawn args   '-Dvisibility=default',
gyp info spawn args   '-Dnode_root_dir=/Users/sergio/Library/Caches/node-gyp/16.0.0',
gyp info spawn args   '-Dnode_gyp_dir=/usr/local/lib/node_modules/npm/node_modules/node-gyp',
gyp info spawn args   '-Dnode_lib_file=/Users/sergio/Library/Caches/node-gyp/16.0.0/<(target_arch)/node.lib',
gyp info spawn args   '-Dmodule_root_dir=/Users/sergio/projects/teams-for-linux/node_modules/spellchecker',
gyp info spawn args   '-Dnode_engine=v8',
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.'
gyp info spawn args ]
gyp info spawn make
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
  CXX(target) Release/obj.target/hunspell/vendor/hunspell/src/hunspell/affentry.o
../vendor/hunspell/src/hunspell/affentry.cxx:544:47: warning: while loop has empty body [-Wempty-body]
                        while (p && *p != ']' && (p = nextchar(p)));
                                                                   ^
../vendor/hunspell/src/hunspell/affentry.cxx:544:47: note: put the semicolon on a separate line to silence this warning
1 warning generated.
  CXX(target) Release/obj.target/hunspell/vendor/hunspell/src/hunspell/affixmgr.o
In file included from ../vendor/hunspell/src/hunspell/affixmgr.cxx:12:
../vendor/hunspell/src/hunspell/affentry.hxx:30:105: warning: implicit conversion of NULL constant to 'unsigned short' [-Wnull-conversion]
  struct hentry *      check_twosfx(const char * word, int len, char in_compound, const FLAG needflag = NULL);
                                                                                                      ~ ^~~~
                                                                                                        0
../vendor/hunspell/src/hunspell/affentry.hxx:93:114: warning: implicit conversion of NULL constant to 'unsigned short' [-Wnull-conversion]
  struct hentry *   check_twosfx(const char * word, int len, int optflags, PfxEntry* ppfx, const FLAG needflag = NULL);
                                                                                                               ~ ^~~~
                                                                                                                 0
../vendor/hunspell/src/hunspell/affixmgr.cxx:3654:65: warning: 'strncmp' call operates on objects of type 'const char' while the size is based on a different type 'const char *' [-Wsizeof-pointer-memaccess]
                             if (strncmp(piece, keyword, sizeof(keyword)) != 0) {
                                                ~~~~~~~         ^~~~~~~
../vendor/hunspell/src/hunspell/affixmgr.cxx:3654:65: note: did you mean to provide an explicit length?
                             if (strncmp(piece, keyword, sizeof(keyword)) != 0) {
                                                                ^~~~~~~
3 warnings generated.
  CXX(target) Release/obj.target/hunspell/vendor/hunspell/src/hunspell/csutil.o
  CXX(target) Release/obj.target/hunspell/vendor/hunspell/src/hunspell/dictmgr.o
  CXX(target) Release/obj.target/hunspell/vendor/hunspell/src/hunspell/filemgr.o
  CXX(target) Release/obj.target/hunspell/vendor/hunspell/src/hunspell/hashmgr.o
In file included from ../vendor/hunspell/src/hunspell/hashmgr.cxx:9:
../vendor/hunspell/src/hunspell/hashmgr.hxx:17:21: warning: private field 'userword' is not used [-Wunused-private-field]
  int               userword;
                    ^
1 warning generated.
  CXX(target) Release/obj.target/hunspell/vendor/hunspell/src/hunspell/hunspell.o
  CXX(target) Release/obj.target/hunspell/vendor/hunspell/src/hunspell/hunzip.o
  CXX(target) Release/obj.target/hunspell/vendor/hunspell/src/hunspell/phonet.o
  CXX(target) Release/obj.target/hunspell/vendor/hunspell/src/hunspell/replist.o
  CXX(target) Release/obj.target/hunspell/vendor/hunspell/src/hunspell/suggestmgr.o
  CXX(target) Release/obj.target/hunspell/vendor/hunspell/src/parsers/textparser.o
  LIBTOOL-STATIC Release/hunspell.a
  CXX(target) Release/obj.target/spellchecker/src/main.o
../src/main.cc:137:25: error: no matching member function for call to 'Set'
      misspelled_range->Set(Nan::New("start").ToLocalChecked(), Nan::New<Integer>(start));
      ~~~~~~~~~~~~~~~~~~^~~
/Users/sergio/Library/Caches/node-gyp/16.0.0/include/node/v8.h:3926:37: note: candidate function not viable: requires 3 arguments, but 2 were provided
  V8_WARN_UNUSED_RESULT Maybe<bool> Set(Local<Context> context,
                                    ^
/Users/sergio/Library/Caches/node-gyp/16.0.0/include/node/v8.h:3929:37: note: candidate function not viable: requires 3 arguments, but 2 were provided
  V8_WARN_UNUSED_RESULT Maybe<bool> Set(Local<Context> context, uint32_t index,
                                    ^
../src/main.cc:138:25: error: no matching member function for call to 'Set'
      misspelled_range->Set(Nan::New("end").ToLocalChecked(), Nan::New<Integer>(end));
      ~~~~~~~~~~~~~~~~~~^~~
/Users/sergio/Library/Caches/node-gyp/16.0.0/include/node/v8.h:3926:37: note: candidate function not viable: requires 3 arguments, but 2 were provided
  V8_WARN_UNUSED_RESULT Maybe<bool> Set(Local<Context> context,
                                    ^
/Users/sergio/Library/Caches/node-gyp/16.0.0/include/node/v8.h:3929:37: note: candidate function not viable: requires 3 arguments, but 2 were provided
  V8_WARN_UNUSED_RESULT Maybe<bool> Set(Local<Context> context, uint32_t index,
                                    ^
../src/main.cc:139:15: error: no matching member function for call to 'Set'
      result->Set(index, misspelled_range);
      ~~~~~~~~^~~
/Users/sergio/Library/Caches/node-gyp/16.0.0/include/node/v8.h:3926:37: note: candidate function not viable: requires 3 arguments, but 2 were provided
  V8_WARN_UNUSED_RESULT Maybe<bool> Set(Local<Context> context,
                                    ^
/Users/sergio/Library/Caches/node-gyp/16.0.0/include/node/v8.h:3929:37: note: candidate function not viable: requires 3 arguments, but 2 were provided
  V8_WARN_UNUSED_RESULT Maybe<bool> Set(Local<Context> context, uint32_t index,
                                    ^
../src/main.cc:223:15: error: no matching member function for call to 'Set'
      result->Set(i, Nan::New(dict.data(), dict.size()).ToLocalChecked());
      ~~~~~~~~^~~
/Users/sergio/Library/Caches/node-gyp/16.0.0/include/node/v8.h:3926:37: note: candidate function not viable: requires 3 arguments, but 2 were provided
  V8_WARN_UNUSED_RESULT Maybe<bool> Set(Local<Context> context,
                                    ^
/Users/sergio/Library/Caches/node-gyp/16.0.0/include/node/v8.h:3929:37: note: candidate function not viable: requires 3 arguments, but 2 were provided
  V8_WARN_UNUSED_RESULT Maybe<bool> Set(Local<Context> context, uint32_t index,
                                    ^
../src/main.cc:249:15: error: no matching member function for call to 'Set'
      result->Set(i, val.ToLocalChecked());
      ~~~~~~~~^~~
/Users/sergio/Library/Caches/node-gyp/16.0.0/include/node/v8.h:3926:37: note: candidate function not viable: requires 3 arguments, but 2 were provided
  V8_WARN_UNUSED_RESULT Maybe<bool> Set(Local<Context> context,
                                    ^
/Users/sergio/Library/Caches/node-gyp/16.0.0/include/node/v8.h:3929:37: note: candidate function not viable: requires 3 arguments, but 2 were provided
  V8_WARN_UNUSED_RESULT Maybe<bool> Set(Local<Context> context, uint32_t index,
                                    ^
../src/main.cc:289:14: error: no matching member function for call to 'Set'
    exports->Set(Nan::New("Spellchecker").ToLocalChecked(), tpl->GetFunction(context).ToLocalChecked());
    ~~~~~~~~~^~~
/Users/sergio/Library/Caches/node-gyp/16.0.0/include/node/v8.h:3926:37: note: candidate function not viable: requires 3 arguments, but 2 were provided
  V8_WARN_UNUSED_RESULT Maybe<bool> Set(Local<Context> context,
                                    ^
/Users/sergio/Library/Caches/node-gyp/16.0.0/include/node/v8.h:3929:37: note: candidate function not viable: requires 3 arguments, but 2 were provided
  V8_WARN_UNUSED_RESULT Maybe<bool> Set(Local<Context> context, uint32_t index,
                                    ^
6 errors generated.
make: *** [Release/obj.target/spellchecker/src/main.o] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:194:23)
gyp ERR! stack     at ChildProcess.emit (node:events:365:28)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:290:12)
gyp ERR! System Darwin 20.5.0
gyp ERR! command "/usr/local/Cellar/node/16.0.0_1/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/sergio/projects/teams-for-linux/node_modules/spellchecker
gyp ERR! node -v v16.0.0
gyp ERR! node-gyp -v v7.1.2
gyp ERR! not ok

```

**Comment by IsmaelMartinez** [comment]
yep, that will be the node version. I think is v10 the last one that that electron supports. You can use nvm to have multiple versions of node if that helps

**Comment by sergiocasero** [comment]
12.22.1 does the job :) Thank you so much! I'll investigate if there is something to change or this is enough...

**Comment by IsmaelMartinez** [comment]
perfecto. I do know that the icon is a bit odd looking but that should not stop the app from working. Over all, it should just work. Do ping if that is not the case and I can see if I can help. Ta

**Comment by sergiocasero** [comment]
Ahhh it works but Unfortunately...  looks like is not enough, login works but it says "You've logged in right, but I can't detect your device model"

**Comment by IsmaelMartinez** [comment]
I wonder if that is a company policy thing. Have you tried logging with a user that is not your company user? aka, hotmail or something?

It could just also be proxy configuration (there is more info about how to configure a proxy in the config folder) . Hope helps

**Comment by IsmaelMartinez** [comment]
maybe check again once the #432 gets merged as the version of node has been increased in there. A superb job by @jijojosephk

**Comment by IsmaelMartinez** [comment]
on that @sergiocasero , you might be able to tweak that by using a windows useragent. 

You can pass the user agent like:

```teams-for-linux --chromeUserAgent "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Teams/1.4.00.22265 Chrome/85.0.4183.121 Electron/10.4.3 Safari/537.36"```

To know what user agent to use, ask one of your colleagues that is running windows:
1. In one channel they own (or that they can add tabs) to add a Tab by clicking 
![image](https://user-images.githubusercontent.com/430992/132052036-51f6c4eb-b9c4-4cb6-b079-5bad5b759c82.png)
2. Then select the type as Website
![image](https://user-images.githubusercontent.com/430992/132052152-68d385eb-82d0-4669-b769-5e60b6730157.png)
3. Put "https://www.whatismybrowser.com/detect/what-is-my-user-agent" or similar 
4. Copy that user agent in place of the one that I put in there.

That might work.

Hope helps

**Comment by jijojosephk** [comment]
@sergiocasero I managed run mac 10.14 on QEMU VM. Application ran without any problem. Then only trouble is the tray icon very big. Other wise login and everything looks good. I'm not quite sure what's the problem you're facing.

**Comment by sergiocasero** [comment]
But the problem in my case is that policies are:
Edge with the company email logged in on Windows. And edge uses native login, I can’t change the user agent on the edge app (as far I know)

**Comment by IsmaelMartinez** [comment]
Check my previous post. The teams-for-linux app does allow you to change
user agent. Steps detailed there.

**Comment by jijojosephk** [comment]
Closing this due to inactivity. Feel free to re-open if needed.

**Comment by IsmaelMartinez** [comment]
Hi All, 

I just had a wee play and you can have it running in macOS. Not sure why but you need to run it without persisting a partition (took me a while to find that out), and there is a wrong icon size in the toolbar:
<img width="342" alt="image" src="https://user-images.githubusercontent.com/430992/193450099-ca3f55be-e38c-4f28-b16b-bb39a5ca46b8.png">

But it does seem to be working. I will be playing with it when I get time (I don't normally have much) but if you run it from source and start with `yarn start --partition=test` it works! (you need to log in each time as the partition is not persisted)

**Comment by IsmaelMartinez** [comment]
<img width="1434" alt="image" src="https://user-images.githubusercontent.com/430992/193451063-aa2b94a4-c132-4d13-8ca3-88ce2db275cd.png">

Seems to be working. I have put a branch with my changes so people can play with it.

https://github.com/IsmaelMartinez/teams-for-linux/tree/feature/wip-add-macos-support

I installed the dmg using that branch and running `yarn dist:macos` but I don't have time to test it.

I fixed the wrong icon, and it looks like the camera ain't working (but I guess that might be permissions). 

I will play with it at some point in a few weeks, as I am offline all next week.

Hopefully I can come back to help a bit even if from MacOS land.
